silverback/_cli.py

```diff
@@ -142,3 +128,500 @@ def run(cli_ctx, account, runner_class, recorder, max_exceptions, path):
+)
+@click.argument("workspace")
+def new_cluster(
+    client: PlatformClient,
+    workspace: str,
+    cluster_name: str | None,
+    cluster_slug: str | None,
+    tier: str,
+    config_updates: list[tuple[str, str]],
+):
+    """Create a new cluster in WORKSPACE"""
+
+    if not (workspace_client := client.workspaces.get(workspace)):
+        raise click.BadOptionUsage("workspace", f"Unknown workspace '{workspace}'")
+
+    configuration = getattr(ClusterTier, tier.upper()).configuration()
+
+    for k, v in config_updates:
+        setattr(configuration, k, int(v) if v.isnumeric() else v)
+
+    cluster = workspace_client.create_cluster(
+        cluster_name=cluster_name,
+        cluster_slug=cluster_slug,
+        configuration=configuration,
+    )
+    click.echo(f"{click.style('SUCCESS', fg='green')}: Created '{cluster.name}'")
+    # TODO: Pay for cluster via new stream
+
+
+# `silverback cluster pay WORKSPACE/NAME --account ALIAS --time "10 days"`
+# TODO: Create a signature scheme for ClusterInfo
+#         (ClusterInfo configuration as plaintext, .id as nonce?)
+# TODO: Test payment w/ Signature validation of extra data
+
+
+@cluster.command(name="info")
+def cluster_info(client: ClusterClient):
+    """Get Configuration information about a CLUSTER"""
+
+    # NOTE: This actually doesn't query the cluster's routes, which are protected
+    click.echo(f"Cluster Version: v{client.version}")
+
+    if config := client.state.configuration:
+        click.echo(yaml.safe_dump(config.model_dump()))
+
+    else:
+        click.secho("No Cluster Configuration detected", fg="yellow", bold=True)
+
+
+@cluster.command(name="health")
+def cluster_health(client: ClusterClient):
+    """Get Health information about a CLUSTER"""
+
+    click.echo(yaml.safe_dump(client.health.model_dump()))
+
+
+@cluster.group()
+def vars():
+    """Manage groups of environment variables in a CLUSTER"""
+
+
+def parse_envvars(ctx, name, value: list[str]) -> dict[str, str]:
+    def parse_envar(item: str):
+        if not ("=" in item and len(item.split("=")) == 2):
+            raise click.UsageError(f"Value '{item}' must be in form `NAME=VAL`")
+
+        return item.split("=")
+
+    return dict(parse_envar(item) for item in value)
+
+
+@vars.command(name="new")
+@click.option(
+    "-e",
+    "--env",
+    "variables",
+    multiple=True,
+    type=str,
+    metavar="NAME=VAL",
+    callback=parse_envvars,
+    help="Environment variable key and value to add (Multiple allowed)",
+)
+@click.argument("name")
+def new_vargroup(client: ClusterClient, variables: dict, name: str):
+    """Create a new group of environment variables in a CLUSTER"""
+
+    if len(variables) == 0:
+        raise click.UsageError("Must supply at least one var via `-e`")
+
+    vg = client.new_variable_group(name=name, variables=variables)
+    click.echo(yaml.safe_dump(vg.model_dump(exclude={"id"})))  # NOTE: Skip machine `.id`
```

Comment:
Yes, please update the CCS field to be unique and "Slug-like"

I didn't want to bother having the separate name and slug fields, seemed anmoying to handle just for the bots (vs. Clusters where it does make sense)

Reply:
> Yes, please update the CCS field to be unique and "Slug-like"

I have no idea what you mean by this. A slug is an ID. So why would we have 2 IDs and a display name? And one ID we can't even use with the API? 

I think we should just use the IDs at this point, since I don't especially want to delay things with yet another refactor. Or if it's just entirely client-side, invent your own way to create slugs and handling name conflicts I guess. Though I also have faith our users can handle UUIDs.

Reply:
A slug is a human-readable ID that is short and serves as an alias to a specific UUID, but is also friendly to CLI/URLs (e.g. don't need quotes or escaping). 

Typing a UUID directly is pretty annoying, really trying to avoid that for the CLI experience. UUIDs are best used as machine-to-machine identifiers that don't change, this is why the service uses UUIDs for it's REST API since any slug/name we give it is metadata that a user should be allowed to change at any time. The CLI uses UUIDs internally to the client but has lets users use the name field instead to select a bot/vg so when you type a command and send some requests, it's always by UUID and not something that can change on you en-route (leading to weird UX issues).

My proposal is that the bot "name" field is treated like a slug, e.g. it is not allowed to contain spaces or special characters like emojis, is case insensitive, globally unique (per cluster), etc. Having a separate "name" field feels redundant for bots, and we could instead add a separate "description" field instead as metadata to give a short blurb to all users who can see it about what the bot is supposed to do.

Lastly, when we go and add UI to the Cluster it will have a route like `https://silverback.apeworx.io/.../b/bot-name` for displaying an overview for a specific bot. That's another reason why the slug is important.

Reply:
No point in repeating myself, so I guess this is resolved. Maybe submit an issue to change the API and I'll get to it when I can.

Reply:
see https://github.com/SilverbackLtd/cluster/issues/101